pywt/_extensions/c/cwt.template.h

```diff
@@ -0,0 +1,66 @@
+/* Copyright (c) 2016 Holger Nahrstaedt */
+/* See COPYING for license details. */
+
+
+#include "templating.h"
+
+#ifndef TYPE
+#error TYPE must be defined here.
+#else
+
+#include "cwt.h"
+
+#if defined _MSC_VER
+#define restrict __restrict
+#elif defined __GNUC__
+#define restrict __restrict__
+#endif
+
+
+TYPE CAT(TYPE, _powof)(const TYPE x, const TYPE y);
+
+
+void CAT(TYPE, _gaus)(const TYPE * const restrict input,
+                              TYPE * const restrict output, const size_t N,
+                              const size_t number);
+
+void CAT(TYPE, _gaus1)(const TYPE  input, TYPE * const restrict output);
```

Comment:
If not inlined, I think it's better to keep these as private functions (i.e. don't exposre them in the header), unless you can think of a reason they would be used elsewhere?